### PR TITLE
chore #29: Refactored auth lib

### PR DESCRIPTION
**Pull request by ktilelis**

### CODE REVIEW

File: libs/shared/auth/src/lib/user-info-container/user-info-container.component.spec.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ import { render, RenderResult, screen } from '@testing-library/angular';
 import { KeycloakService } from 'keycloak-angular';
 import { KeycloakProfile } from 'keycloak-js';
 import { provideMockStore } from '@ngrx/store/testing';
-import { AuthSelectors } from '@expense-tracker-ui/shared/auth/data-access';
+import { AuthSelectors } from '../../index';
+import '@testing-library/jest-dom';
```
Comment (atrifyllis) [inline libs/shared/auth/src/lib/user-info-container/user-info-container.component.spec.ts:8]: why is this needed?

File: libs/shared/auth/src/lib/user-info/user-info.component.ts [hunk 2]
```diff
@@ -13,6 +13,6 @@ export class UserInfoComponent {
   @Input() isLoggedIn: boolean | null = false;
   @Input() userProfile: KeycloakProfile | null = null;
 
-  @Output() login = new EventEmitter<any>();
-  @Output() logout = new EventEmitter<any>();
+  @Output() login = new EventEmitter<void>();
```
Comment (atrifyllis) [inline libs/shared/auth/src/lib/user-info/user-info.component.ts:16]: were there compile errors?

**Review by atrifyllis — APPROVED** [review]

File: libs/shared/auth/src/lib/user-info/user-info.component.ts (hunk 2 above)
Comment (ktilelis) [reply libs/shared/auth/src/lib/user-info/user-info.component.ts:16]: There were not. I just thought that void is more representative of the fact that the emitter carries no payload

File: libs/shared/auth/src/lib/user-info-container/user-info-container.component.spec.ts (hunk 1 above)
Comment (ktilelis) [reply libs/shared/auth/src/lib/user-info-container/user-info-container.component.spec.ts:8]: For some reason expect(...).toBeInTheDocument() was not being recognized. Adding the import statement fixed the issue.
It's strange though...
